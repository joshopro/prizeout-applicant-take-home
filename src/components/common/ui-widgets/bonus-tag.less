@import '../../../utils/styles/mixins.less';

.bonus-tag {
    .flexbox(@justifyContent: space-between);

    &.bonus-tag--small {
        .bonus-tag__container {
            height: 1.25rem;

            &::before {
                border-right-width: .625rem;
                border-left-width: 0;
                border-top-width: .625rem;
                border-bottom-width: .625rem;
                left: unset;
                border-right-color: transparent;
                right: -.625rem;
            }
        }
    }

    &__container {
        .flexbox(@justifyContent: space-between);
        background-color: var(--color-blue-700);
        color: var(--color-white);
        padding: 0 0.5rem;
        height: 1.875rem;
        border-radius: 0;
        box-sizing: border-box;
        position: relative;
    
    
        &::before {
            content: "";
            position: absolute;
            display: block;
            top: 0;
            border: .625rem solid var(--color-blue-700);
            border-right-width: .625rem;
            border-left-width: 0;
            border-top-width: 0.9375rem;
            border-bottom-width: 0.9375rem;
            left: unset;
            border-right-color: transparent;
            right: -.625rem;
        }
    
        > div {
            .flexbox(@justifyContent: flex-start);
    
            & > i {
                width: 1.125rem;
                margin-right: .675rem;
            }
    
            & > small {
                line-height: unset;
            }
        }
    
    }

    &.monetary {
        .bonus-tag__container {
            background-color: var(--color-green-400);

            &::before {
                border-color: var(--color-green-400);
                border-right-color: transparent;
            }
        }
    }
}