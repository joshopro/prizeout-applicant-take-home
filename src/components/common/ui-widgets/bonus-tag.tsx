import React from 'react';
import Classnames from 'classnames';
import PropTypes from 'prop-types';

import './bonus-tag.less';

interface BonusTagProps {
    size?: string;
    type: string;
    value: number;
}

export const BonusTag: React.FC<BonusTagProps> = ({ type, value, size }): React.ReactElement => {
    const classes: string = Classnames(
        `bonus-tag`,
        { monetary: type === 'monetary' },
        { [`bonus-tag--${size}`]: size },
    );
    const bonus = type === 'monetary' ? `$${value}` : `${value}%`;

    return (
        <div className={classes}>
            <div className="bonus-tag__container">
                <div>
                    <i className="fak fa-prizeout-bonus-icon" />
                    <small>{bonus}</small>
                </div>
            </div>
        </div>
    );
};

BonusTag.propTypes = {
    size: PropTypes.string,
    type: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
};
