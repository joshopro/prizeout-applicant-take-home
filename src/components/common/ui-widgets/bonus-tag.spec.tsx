import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { BonusTag } from './bonus-tag';

describe('Test Bonus Tag Component', () => {
    test('Component matches snapshot', () => {
        const { asFragment } = render(<BonusTag type="percentage" value={15.0} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Monetary bonus displayed correctly', () => {
        const { container } = render(<BonusTag type="monetary" value={22.2} />);

        expect(screen.queryByText('$22.2')).toBeInTheDocument();
        expect(container.getElementsByClassName('monetary').length).toBe(1);
    });

    test('Percentage bonus displayed correctly', () => {
        const { container } = render(<BonusTag type="percentage" value={15.1} />);

        expect(screen.queryByText('15.1%')).toBeInTheDocument();
        expect(container.getElementsByClassName('monetary').length).toBe(0);
    });
});
