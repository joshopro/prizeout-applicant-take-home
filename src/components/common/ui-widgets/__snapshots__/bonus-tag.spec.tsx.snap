// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Bonus Tag Component Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bonus-tag"
  >
    <div
      class="bonus-tag__container"
    >
      <div>
        <i
          class="fak fa-prizeout-bonus-icon"
        />
        <small>
          15%
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
